src/controllers/user.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const { userService } = require('../service/user.js');
+
+const add = async(req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.sendStatus(400);
+
+    return;
+  };
+
+  const newUser = await userService.create(name);
+
+  res.statusCode = 201;
+  res.send(newUser);
+};
+
+const getAll = async(req, res) => {
+  const users = await userService.getAll();
+
+  if (!users.length) {
+    res.send([]);
+
+    return;
+  }
+
+  res.send(users);
+};
+
+const getOne = async(req, res) => {
+  const userId = +req.params.userId;
+
+  if (typeof userId !== 'number') {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const foundUser = await userService.getById(userId);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.sendStatus = 200;
+  res.send(foundUser);
+};
+
+const update = async(req, res) => {
+  const userId = +req.params.userId;
+  const { name } = req.body;
+
+  if (typeof userId !== 'number') {
+    res.sendStatus(400);
+
+    return;
+  }
```

Comment:
i have doubts about this check in case of update

Reply:
I do this check, because in my previous work without it tests failed...